v2/observer/jobs.go

```diff
@@ -222,6 +222,17 @@ func (o *observer) startJobPodTimer(ctx context.Context, pod *corev1.Pod) {
 			)
 			return
 		}
+		if timeout > o.config.maxJobLifetime {
+			o.errFn(
+				fmt.Errorf(
+					"timeout %q for pod %q exceeds the configured maximum %q",
+					duration,
+					pod.Name,
+					o.config.maxJobLifetime,
+				),
+			)
+			return
+		}
```

Comment:
I had been thinking that if the pod-specified timeout exceeded the max, we'd just reduce the timeout to the max.

If we return an error here, don't we prevent this pod from _ever_ timing out?

Reply:
Ah, yup, as you originally requested, but I missed :)

Updated the logic to fallback to the config'd max if we can't parse the pod duration (including if it is unset/empty) or it exceeds the max.  We log an error in this case, for tracking in the observer logs.